### ISSUE TITLE
Gtk can't load icons

### ISSUE THREAD

**Issue by observer1** [picoe/Eto#603]
In version 2.3 I'm seeing this error
GLib.GException: Compressed icons are not supported
when using
Eto.Drawing.Icon.FromResource

In both gtk2 and gtk3

this works in 2.2

**Comment by cwensley** [comment]
Thanks for reporting the issue!  Do you have more information such as what type of icon you are trying to load and potentially an example file that can reproduce this?

**Comment by observer1** [comment]
I have a test icon but I can't find a way to attach it.
It can be made using imagemagick from an svg file with this command:

`convert -background transparent "$FILE" -define icon:auto-resize "${FILE%%.*}".ico`

**Comment by cwensley** [comment]
To attach a file, drag it or click the 'selecting them' link below where you enter the comment (;

**Comment by observer1** [comment]
It doesn't accept icons, here is zipped

[Untitled-1.ico.zip](https://github.com/picoe/Eto/files/601866/Untitled-1.ico.zip)

**Comment by observer1** [comment]
I just found it's because the first image in the icon is png removing it works

**Comment by cwensley** [comment]
ah, that makes sense.. glad you got it working!  shall we close the issue?

**Comment by observer1** [comment]
I don't know but I think is a bug specially because it worked in previous versions and it works in winforms

**Comment by cwensley** [comment]
Hm, that's a good point.  It appears it's because with the changes in 2.3 for High DPI it now uses Gtk.IconSet so you can access each of the bitmaps in the icon.  Interestingly, we may be able to compose the IconSet manually which we already do.  

I have no idea whether all other platforms support compressed images in ico files, though if they do getting GTK to support it as well would be nice.

**Comment by observer1** [comment]
It works in wpf, and it doesn't fail in xammac2 but I'm not sure what does there because windows don't have an icon and I haven't used icons in other controls yet

**Comment by cwensley** [comment]
@observer1 cool, thanks for letting me know.  I don't have time to do this right now but if you are feeling up to it Gtk's [IconHandler.cs](https://github.com/mcneel/Eto/blob/develop/Source/Eto.Gtk/Drawing/IconHandler.cs) probably needs the CreateFrames() to also do something similar to the Create() method to build the IconSet instead of creating it from the Pixbuf.

**Comment by cra0zy** [comment]
@cwensley Feel free to ping me to investigate Gtk related issues.

**Comment by cra0zy** [comment]
Well... turns out that Gtk does not support compressed images, the reason code worked in Eto 2.2.0 is because if you try to load an ico file with Gdk.Pixbuf, it will try to load the biggest uncompressed image from it.

**Comment by darkdragon-001** [comment]
Did someone report this [upstream](https://gitlab.gnome.org/GNOME/gtk/-/issues)?

**Comment by darkdragon-001** [comment]
I am experiencing a similar problem with mono itself. See mono/mono#19673.

**Comment by harry-cpp** [comment]
Are you by any chance using Linux with dotnet installation from snap? If so, snap dotnet is sandboxed.

**Comment by darkdragon-001** [comment]
No, I use the deb version of dotnet. Furthermore, for this project I am using mono since it requires WPF. What does this have to do with the bug anyways?

**Comment by Shoxx98** [comment]
also want this feature to be implemented
